app/bundles/CampaignBundle/EventListener/LeadSubscriber.php

```diff
@@ -99,7 +99,7 @@ public function onLeadListBatchChange(ListChangeEvent $event)
 
                     $removeLeads = [];
                     foreach ($leads as $l) {
-                        $lists = (isset($leadLists[$l])) ? $leadLists[$l] : [];
+                        $lists = (isset($leadLists[$l['id']])) ? $leadLists[$l['id']] : [];
```

Comment:
This would mean that line 82 is bad as well because getLeadLists() is expecting $leads to be a flat array of IDs. https://github.com/mautic-inc/mautic/blob/master/app/bundles/LeadBundle/Entity/LeadListRepository.php#L172

Reply:
I do not think L82 is bad because getLeadLists() do a Doctrine query with $leads and the query support collection as parameter.
https://www.doctrine-project.org/projects/doctrine-orm/en/2.5/changelog/migration_2_5.html#query-api-where-in-query-using-a-collection-as-parameter

Reply:
@alanhartless what do you think of that answer?

Reply:
I didn't realize that for Doctrine. Explains why it seems to work today. I assume that Doctrine handles an array of arrays as well since it's not an array of entities (Lead[])

Reply:
Doctrine query like `->where('u.foo IN (:foo)')` handles :array of id, array of object entities `$qb->getQuery()->getResult()` and array of array entites `$qb->getQuery()->getArrayResult()`.